src/_posts/databases/mysql/2000-01-01-mysql-8-prerequisites.md

```diff
@@ -69,3 +70,129 @@ In this situation, you need to add a primary key to the `foo` table. You need to
 ```sql
 ALTER TABLE 'foo' ADD COLUMN <column description> PRIMARY KEY;
 ```
+
+## One Command Script
+
+Below you can find a script that can be run directly on the database to:
+- create missing primary keys for the concerned tables
+- set the engine to InnoDB for the concerned tables
+
+In fact, the script create two procedures, one for each previous actions.
+
+{% warning %}
+  As always, we highly recommend to make a backup before executing the script.
+{% endwarning %}
+
+```sql
+DELIMITER $$
+DROP PROCEDURE IF EXISTS `create_missing_pk`$$
+
+CREATE PROCEDURE `create_missing_pk`()
+BEGIN
+
+DECLARE v_finished INTEGER DEFAULT 0;
+DECLARE v_table VARCHAR(100) DEFAULT "";
+DECLARE stmt VARCHAR(500) DEFAULT "";
+
+-- get the list of tables without PK
+DECLARE table_cursor CURSOR FOR
+SELECT i.table_name \
+  FROM information_schema.tables AS i LEFT JOIN information_schema.key_column_usage AS c ON (\
+    i.table_name = c.table_name AND \
+    c.constraint_schema = i.table_schema AND \
+    c.constraint_name = 'PRIMARY' \
+  ) \
+  WHERE i.table_schema \
+    NOT IN ('information_schema', 'performance_schema', 'sys', 'mysql', 'mysql_innodb_cluster_metadata') AND \
+    i.table_type != "VIEW" AND \
+    c.constraint_name IS NULL;
+
+DECLARE CONTINUE HANDLER
+FOR NOT FOUND SET v_finished = 1;
+
+OPEN table_cursor;
+
+-- loop over all tables without PK
+alter_tables: LOOP
+
+    FETCH table_cursor INTO v_table;
+    IF v_finished = 1 THEN
+    LEAVE alter_tables;
+    END IF;
+
+    -- below the alter table which create PKs
+    SET @prepstmt = CONCAT('ALTER TABLE ',v_table,' ADD COLUMN id INT AUTO_INCREMENT NOT NULL PRIMARY KEY FIRST;');
```

Comment:
**issue:** this column name is too common. for the record, I have a table that have a column named `id` but not primary. This query fails to execute.

```suggestion
    SET @prepstmt = CONCAT('ALTER TABLE ',v_table,' ADD COLUMN generated_pk INT AUTO_INCREMENT NOT NULL PRIMARY KEY FIRST;');
```